### PR TITLE
Add AWS SES dedicated IP warm-up functionality

### PR DESCRIPTION
**Pull request by SpootDev**
## Summary
This PR adds support for AWS SES dedicated IP warm-up, implementing the AWS-recommended gradual volume ramp schedule to establish sender reputation with mailbox providers.

## Key Changes
- **New module**: `mk_lib_network_email_aws_warmup` with AWS SES warm-up implementation
  - `WARMUP_PLAN`: 45-day schedule following AWS's published recommendations, ramping from 50 to 30,000,000 daily sends
  - `mk_lib_network_email_aws_warmup_daily_target()`: Returns the daily send target for a given warm-up day, with clamping for days past the schedule
  - `WarmupConfig`: Configuration struct for a warm-up batch run (AWS credentials, sender details, day number, rate limit, messages)
  - `WarmupReport`: Result struct tracking attempted/sent/failed counts and errors
  - `mk_lib_network_email_aws_warmup_run()`: Async function that executes a warm-up batch, pacing sends to a configurable rate limit using `spawn_blocking` to avoid blocking the async runtime
- **Module export**: Added `mk_lib_network_email_aws_warmup` to `lib.rs`

## Implementation Details
- The warm-up schedule is indexed by 1-based day (day 1 = `WARMUP_PLAN[0]`)
- Send rate is controlled via `max_per_second` with calculated intervals between sends
- Each message is sent via `tokio::task::spawn_blocking` to keep blocking I/O off the async runtime
- Comprehensive error handling with per-message error tracking (invalid addresses, send failures, task join errors)
- Full test coverage including edge cases (day 0, day 1, past-schedule clamping, monotonicity validation)

https://claude.ai/code/session_016TdtPhgESBXE9akiG95RHw

### CODE REVIEW

File: src/mk_lib_network/src/mk_lib_network_email_aws_warmup.rs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        let mailer_clone = mailer.clone();
+        let send_res = tokio::task::spawn_blocking(move || mailer_clone.send(&email)).await;
+
+        match send_res {
+            Ok(Ok(_)) => report.sent += 1,
+            Ok(Err(e)) => {
+                report.failed += 1;
+                report.errors.push(format!("send failed: {e}"));
+            }
```
[the remark this answers is not in the document]
Comment (SpootDev) [reply src/mk_lib_network/src/mk_lib_network_email_aws_warmup.rs]: Good catch — fixed in e0c1aa5c. Added `WARMUP_MAX_ERROR_SAMPLES = 100` and a `WarmupReport::record_failure` helper that increments `failed` on every failure but only pushes to `errors` until the cap. Also added a unit test covering the cap.

---
_Generated by [Claude Code](https://claude.ai/code)_
